Feat/system exits

mlfurman3 opened the pull request:
Creates System Exits tab within new System Performance Menu with 3 subtabs:

<img width="171" height="129" alt="image" src="https://github.com/user-attachments/assets/c7eada9a-63ec-43cc-8ce6-c6d4228becd3" />

1. System Exit Types - treemap of destination type for system exits
2. System Exit Comparisons
    a. Time Chart - dotplot of destination type for current year of reporting and previous year of reporting
    b. Subpopulation Chart - dotplot of destination
3.  System Exit Demographics - 1D and 2D heatmaps with demographic checkboxes, just like System Comp ones 
    a. I used `conditionalPanels` to switch between the 1D and 2D heatmaps, and a `renderCachedPlot` to address the chart resizing issues.

Each subtab has their own Data and Image Download buttons. The Types and Comparisons charts should be responsive to all of the filters above; the Demographics ones are only responsive to the top row of filters (just like system overview).

I tried to generalize some of the code being re-used from system_overview / system_composition.R into a more general `system_performance_functions.R`. The system exits-specific stuff is in a new `system_exits_server.R`. Let me know if reorganizing or renaming things would make more sense.

mlfurman3 committed:
refactor system performance heatmap into separate functions for system comp and system exits, since exits needs percentages and different underlying dataset

mlfurman3 committed:
fix CSS for 1D system exits heatmap to prevent stretching chart image

mlfurman3 committed:
add Excel download functionality for system exits comparisons tab, using placeholder data for now

mlfurman3 committed:
incorporate system exit check for days_to_lookahead > 14, filter out syse enrollment filter selections

mlfurman3 committed:
create tree_exits_data reactive for use in system exits by type treemap, data exports. still need to add crosswalk for destination reference numbers and labels

mlfurman3 committed:
update system exits by type treemap function to use new reactive variables, add argument to optionally show legend

mlfurman3 committed:
update destination counts and labels in system exit by types data and image downloads

mlfurman3 committed:
update data used in system exits subpopulation and time charts to pull in syse filters and reactive data frames. also add validation checks on # of rows

mlfurman3 committed:
update header info above system exits time chart to show 2-year report period instead of 1-year

mlfurman3 committed:
update renv lockfile with renv::snapshot

mlfurman3 committed:
Merge branch 'feat/monthly-inflow-outflows' into feat/system-exits

Merge branch 'feat/monthly-inflow-outflows' into feat/system-exits

# Conflicts:
#	hardcodes.R
#	system_composition_server.R
#	system_inflow_outflow_server.R
#	system_overview_server.R
#	ui.R

mlfurman3 committed:
update system comp reactive variable names from latest MbM merge, add re-rendering to 1D system exits heatmap when subtab changes

mlfurman3 committed:
add re-rendering of 1D system comp and system exit heatmaps when app tab changes (id 'pageid')

mlfurman3 committed:
update shinytests from MbM-exits merge changes

mlfurman3 committed:
add flextable functions for system exits subpopulation and time tables - used for table generation in PPT download

mlfurman3 committed:
apply conditional theming to system exits charts with tables below - for app, use table headers and hide x-axis text; for PPT download, do the opposite

mlfurman3 committed:
add initial PPT download for system exits comparison subtab, with slides for subpop chart and table, time chart and table

mlfurman3 committed:
add autofit to system exits flextables for wider appearance in PPT slides

mlfurman3 committed:
update system exits logic to use inflow-outflow functions - work in progress, needs further tweaking. it currently matches inflow-outflow detail chart counts

mlfurman3 committed:
switch order of time and subpopulation charts in System Exits Comparison subtab

mlfurman3 committed:
fix methodology_type bug in system exits PHD heatmap code

mlfurman3 committed:
create equivalent system exits functions for get_lookbacks, get_eecr_and_lecr, and expand_by_periods

mlfurman3 committed:
revise detail box for system exits Time Chart subtab to show Previous and Current Year dates

mlfurman3 committed:
add all_filtered_syse_time reactive for data specific to time chart and 2-year reporting period. may need to adjust reactivity so it responsive to filter changes when on subpopulation tab

mlfurman3 committed:
create System Exits Comparisons data for data download by destination type and destination type detail. optionally include Subpopulation sheet if the demographic filters have been modified

mlfurman3 committed:
add space between system exits by type title and chart

mlfurman3 committed:
edits to System Exits by Type data download - added Total row in metadata, percent column in data

mlfurman3 committed:
add list_all_destinations helper function for summarizing system exits by destination and dest type detail with all possible levels

mlfurman3 committed:
add additional destination levels to system exits by type data download, fill with 0s

mlfurman3 committed:
refactor system exits subpopulation chart logic - use everyone_else reactive df, add rows for totals by destination type to data download

mlfurman3 committed:
refactor system exits time chart logic - use everyone reactive, add rows for totals by destination type to data download

mlfurman3 committed:
add total counts to system exits PH demog metadata - total people with System Exit, PH System Exit

mlfurman3 committed:
adjust Excel tab naming for System Exits PH Demographics 2D heatmaps

mlfurman3 committed:
build export data frame for system exits PHD 2D heatmap data downloads - sorting, renaming, adding suppression flag

mlfurman3 committed:
add req statements to 2 system exit reactives

mlfurman3 committed:
add arg to suppress_values function to optionally keep the original counts as a column n_orig

mlfurman3 committed:
for 2D system exit heatmaps, export unsuppressed count values

mlfurman3 committed:
for 1D system exit heatmaps, export unsuppressed count values

mlfurman3 committed:
correct date year mismatch in system exits time chart header

mlfurman3 committed:
fix logic for hiding demog filters in sys overview and sys exits heatmap tabs. the header names needed updating to include the h4 tags

mlfurman3 committed:
remove destination column after filtering for PH demog in system exit heatmaps. it was causing an unlabeled extra row to be added to the data downloads

mlfurman3 committed:
adjust heatmap fill levels and breaks for system exits PHD charts

mlfurman3 committed:
update system exit comparisons data download to have separate metadata for Time and Subpop charts, add rows with total counts

mlfurman3 committed:
standardize System Exit download filenames

mlfurman3 committed:
split system exit demographics heatmaps into 2 separate outputs, with conditionalPanels to show/hide each of them depending on how many selections are checked. this prevents the resizing of the 1D plots when you go from 0 to 1 selection

mlfurman3 committed:
disconnect system exit demographics heatmaps from bottom row of filters. should now only respond to changes in top row

mlfurman3 committed:
cache system exit 2D demographic heatmap with renderCachedPlot to avoid rendering delays due to resizing and pre-used filter combinations

mlfurman3 committed:
code cleanup - remove unused system comp if statements from system exit phd chart functions

mlfurman3 committed:
add total counts to system exit time chart metadata in second case - when no subpopulation is selected

mlfurman3 committed:
Merge MbM into feat/system-exits, resolve logic conflicts for lookbacks and lecr functions

Merge branch 'feat/monthly-inflow-outflows' into feat/system-exits

# Conflicts:
#	hardcodes.R
#	helper_functions.R
#	renv.lock
#	renv/activate.R
#	system_composition_server.R
#	system_overview_server.R
#	tests/testthat/_snaps/linux-4.4/main-valid/test-main-valid-sys-flow-mbm-w-AO-Residential-PH.json

mlfurman3 committed:
revise get_was_lh_info to work for 2-year report period in system exits time chart

mlfurman3 committed:
refactor get_was_lh_info_time to use long dataset and two sets of report dates, grouped instead of looped

mlfurman3 committed:
unbold text labels in system exits by type treemap

mlfurman3 committed:
adjust aspect ratio of system exits by type treemap

mlfurman3 committed:
improvements to system exit comparisons data and image downloads - separate metadata slides, labels for fields, add total row counts, conditional subpop slide inclusion

mlfurman3 committed:
system exit demog heatmaps: fix 1D ppt download, resize text in PPT images to fit within cells

mlfurman3 committed:
reword slide titles in system exit demographics PPT download

mlfurman3 committed:
add second line to title of System Exit Demographics charts to indicate PH exit count

mlfurman3 committed:
renv updates

mlfurman3 committed:
adjust colors of System Exit Types treemap to highlight Permanent and Homeless categories

mlfurman3 committed:
add black borders between system exit treemap sections for easier visibility

mlfurman3 committed:
update System Exit Subpopulation chart -  colors, labels, point size, arrow size

mlfurman3 committed:
update System Exit Time chart - colors, labels, point size, arrow size

mlfurman3 committed:
update labels for System Exit Demographics summary and PPT download

mlfurman3 committed:
Merge branch 'feat/system-inflow-outflows' into 'feat/system-exits'

Conflicts: hardcodes.R, deleted shinytests json files

Merge branch 'feat/monthly-inflow-outflows' into feat/system-exits

# Conflicts:
#	hardcodes.R
#	tests/testthat/_snaps/linux-4.4/main-valid/test-main-valid-just-uploaded.json
#	tests/testthat/_snaps/linux-4.4/main-valid/test-main-valid-sys-comp-all-re-veteran.json
#	tests/testthat/_snaps/linux-4.4/main-valid/test-main-valid-sys-comp-all-re.json
#	tests/testthat/_snaps/linux-4.4/main-valid/test-main-valid-sys-comp-chart-default.json
#	tests/testthat/_snaps/linux-4.4/main-valid/test-main-valid-sys-comp-information.json
#	tests/testthat/_snaps/linux-4.4/main-valid/test-main-valid-sys-flow-detail-w-AO-Residential-PH.json
#	tests/testthat/_snaps/linux-4.4/main-valid/test-main-valid-sys-flow-detail-w-AO-Residential.json
#	tests/testthat/_snaps/linux-4.4/main-valid/test-main-valid-sys-flow-detail.json
#	tests/testthat/_snaps/linux-4.4/main-valid/test-main-valid-sys-flow-fth-w-AO-Residential-PH.json
#	tests/testthat/_snaps/linux-4.4/main-valid/test-main-valid-sys-flow-inactive-w-AO-Residential-PH.json
#	tests/testthat/_snaps/linux-4.4/main-valid/test-main-valid-sys-flow-mbm-w-AO-Residential-PH.json
#	tests/testthat/_snaps/linux-4.4/main-valid/test-main-valid-sys-flow-summary-w-AO-Residential-PH.json
#	tests/testthat/_snaps/linux-4.4/main-valid/test-main-valid-sys-flow-summary.json
#	tests/testthat/_snaps/linux-4.4/main-valid/test-main-valid-sys-status-chart.json
#	tests/testthat/_snaps/linux-4.4/main-valid/test-main-valid-sys-status-information.json

mlfurman3 committed:
fix in_dev_mode merge conflict

mlfurman3 committed:
cleanup unused browser statements

mlfurman3 committed:
adjust system exit time and subpop chart x-axis width for PPT download to make axis labels readable.

mlfurman3 committed:
adjust cell border widths for system exit comparisons time and subpop tables - both DT in-app tables and flextable for PPT downloads

mlfurman3 committed:
fix flextable border pipe chain, in_dev_mode typo

mlfurman3 committed:
add ggarchery package to adjust arrow lengths in system exits comparison charts - geom_arrowsegment and position_attractsegment functions

mlfurman3 committed:
adjust x-position placement of circles and arrows in system exits Time and Subpopulation charts. the previous positions were not well aligned with table headers below

mlfurman3 committed:
Merge feat/monthly-inflow-outflows into feat/system-exits

Merge branch 'feat/monthly-inflow-outflows' into feat/system-exits
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

mlfurman3 committed:
remove unused button + logic for system exit client-level download

mlfurman3 committed:
Merge feat/monthly-inflow-outflows into feat/system-exits

Merge branch 'feat/monthly-inflow-outflows' into feat/system-exits
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

mlfurman3 committed:
shinytests - update main_test_script with System Exit tab click-throughs and downloads, re-generate all tests

mlfurman3 committed:
update yellow points (#FCB248) in System Exits comparison charts to purple (#D5BFE6)

alex-silverman reviewed (changes requested):
some non-critical changes, but otherwise looks good. nice job renaming variables (e.g. syso to sys).

mlfurman3 committed:
move system demographics heatmap functions back to system_comp and sytem_exits server scripts

mlfurman3 committed:
replace dplyr code with collapse in system demographics plotting function

mlfurman3 committed:
add descriptions for all libraries loaded in global.R

mlfurman3 committed:
replace data.frame with data.table in list_all_destinations helper function

mlfurman3 committed:
Convert bulk of system exits dplyr code to collapse - remaining pieces involve more complicated column name evaluation

mlfurman3 committed:
remove get_syse_lookbacks function as it is no longer used

mlfurman3 committed:
combine expand_by_periods and expand_by_periods_syse functions into single function

Commit:
grouping the Demographic chart functions together. the others cut across all System PErformance tabs and charts. No changes to the underlying code.

alex-silverman reviewed (changes requested):
functionally it's good to go, but I think it's worth cleaning up the code now. As we discussed, there are a few System Exit functions that are very similar to their System Overview counterparts, ecept for this "time" variable. If we can combine, that would be great. In general, if we're doing the same thing in multiple places, we should try to combine. Of course, not worth it in some cases.

Made one other comment where you get days_since_lookback and determine  continuous_at_end and exits, as the logic has changed slightly. You can see how it's done in the System Overview script.

mlfurman3 committed:
move expand_by_periods function to system_performance_functions.R since it is used in both sys overview and sys exits reactives

mlfurman3 committed:
combine two detailBox functions into sinle sys_detailBox, add detail_type argument for time chart case

mlfurman3 committed:
remove extra column in 2D sys exits heatmap join that caused warning. output remains the same

mlfurman3 committed:
rename sys_perf_selection_info to sys_demographic_selection_info for more specific naming

mlfurman3 committed:
Merge feat/monthly-inflow-outflows into feat/system-exits

Merge branch 'feat/monthly-inflow-outflows' into feat/system-exits
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

mlfurman3 committed:
remove redundant get_syse_eecr_and_lecr function, works the same as get_eecr_and_lecr

mlfurman3 committed:
add get_days_to_next_lh to system exit reactives, pull out re-used code into separate enrollments_last_lh_syse reactive

mlfurman3 committed:
update definitions for continuous_at_end, exited_system in system exits reactives

mlfurman3 committed:
remove extra comma in fmutate statement

mlfurman3 committed:
combine get_was_lh_info and get_was_lh_info_time functions - adds ReportStart, ReportEnd columns that vary depending on time_chart arg

mlfurman3 commented on system_performance_functions.R:
```diff
@@ -0,0 +1,1209 @@
+  
+}
+
+get_was_lh_info_time <- function(period_enrollments_filtered, all_filtered) {
```
These are now combined, with a time_chart boolean arg. Added the ReportStart and ReportEnd columns to the df inside the function

alex-silverman commented on helper_functions.R:
```diff
@@ -581,4 +590,43 @@ get_all_enrollments_for_debugging <- function(bad_records, universe_w_ppl_flags,
+  }
+  
+  ## assumes you are passing a df with columns for Destination Type (Homeless, Temporary, Permanent, Institutionl, Other/Unknown)
+  joined_df <- left_join(destinations_df, df, 
```
convert to collapse 
```suggestion
  joined_df <- join(destinations_df, df, on=c('Destination Type','Destination Type Detail')
```

alex-silverman commented on system_performance_functions.R:
```diff
@@ -0,0 +1,875 @@
+        names_from = selections[1],
+        values_from = n,
+        values_fill = list(n = 0)
+      )
```
if this code goes pretty quickly, then ignore. Otherwise, can look into converting to `collapse` `pivot()` function.

alex-silverman reviewed (changes requested):
okay a couple last changes. Sorry if these seem so nitpicky. Once you make these, go ahead and merge!

alex-silverman commented on system_performance_functions.R:
```diff
@@ -0,0 +1,1209 @@
+  return(l1)
+}
+
+syse_time_detailBox <- function(
```
left a comment on requested change to combine. not a necessity.

mlfurman3 committed:
convert list_all_destinations helper function to collapse

Co-authored-by: alex-silverman <alexsilverman96@gmail.com>

mlfurman3 committed:
convert list_all_destinations join to collapse

Co-authored-by: alex-silverman <alexsilverman96@gmail.com>

mlfurman3 committed:
fix typo in list_all_destinations helper function

mlfurman3 committed:
convert heatmap pivot code to collapse version

mlfurman3 committed:
remove redundant reactive for syse_detailbox

mlfurman3 committed:
Update sys_detailbox function to work for all variations of sys overview pages

Co-authored-by: alex-silverman <alexsilverman96@gmail.com>

mlfurman3 committed:
incorporate new sys_detailbox function and update arguments wherever it is used

mlfurman3 committed:
Merge branch 'feat/monthly-inflow-outflows' into feat/system-exits

mlfurman3 committed:
fix various system comp/exits syntax errors identified during shinytests

mlfurman3 committed:
run main-valid shinytests with latest system exits branch

mlfurman3 commented:
@alex-silverman made the suggested edits, as well as merged in latest MbM branch, fixed a couple of minor syntax issues, and updated shinytests. Should be in good shape but I believe we want to keep it separate from MbM for now

alex-silverman reviewed (approved):
Approved so we know it's good to go. waiting on MbM first

mlfurman3 committed:
Merge latest MbM into 'feat/system-exits'

Merge branch 'feat/monthly-inflow-outflows' into feat/system-exits

# Conflicts:
#	global.R
#	helper_functions.R
#	system_inflow_outflow_server.R
#	system_overview_server.R
#	tests/testthat/_snaps/linux-4.5/main-valid/test-main-valid-exportTestValues.json
#	tests/testthat/_snaps/linux-4.5/main-valid/test-main-valid-sys-comp-all-re.json
#	tests/testthat/_snaps/linux-4.5/main-valid/test-main-valid-sys-flow-detail.json
#	tests/testthat/_snaps/linux-4.5/main-valid/test-main-valid-sys-flow-mbm-w-AO-Residential-PH.json
#	tests/testthat/_snaps/linux-4.5/main-valid/test-main-valid-sys-flow-summary.json

mlfurman3 committed:
update expand_by_periods function with latest MbM logic, house in system_performance_functions.R

mlfurman3 committed:
Initial merge of dev into system exits. Will likely need further reworking

Merge branch 'dev' into feat/system-exits

# Conflicts:
#	hardcodes.R
#	helper_functions.R
#	system_composition_server.R
#	tab_instructions.R
#	tests/testthat/_snaps/linux-4.5/fsa/test-fsa-001.json
#	tests/testthat/_snaps/linux-4.5/fsa/test-fsa-002.json
#	tests/testthat/_snaps/linux-4.5/fsa/test-fsa-003.json
#	tests/testthat/_snaps/linux-4.5/fsa/test-fsa-004.json
#	tests/testthat/_snaps/linux-4.5/fsa/test-fsa-005.json
#	tests/testthat/_snaps/linux-4.5/main-valid/test-main-valid-exportTestValues.json
#	tests/testthat/_snaps/linux-4.5/main-valid/test-main-valid-just-uploaded.json
#	tests/testthat/_snaps/linux-4.5/main-valid/test-main-valid-sys-flow-detail-w-AO-Residential-PH.json
#	tests/testthat/_snaps/linux-4.5/main-valid/test-main-valid-sys-flow-summary-w-AO-Residential-PH.json
#	tests/testthat/_snaps/linux-4.5/main-valid/test-main-valid-sys-status-chart.json
#	tests/testthat/_snaps/linux-4.5/main-valid/test-main-valid-sys-status-information.json
#	tests/testthat/_snaps/linux-4.5/missing-export/test-missing-export-001.json
#	tests/testthat/_snaps/linux-4.5/missing-export/test-missing-export-002.json
#	tests/testthat/_snaps/linux-4.5/missing-export/test-missing-export-003.json
#	tests/testthat/_snaps/linux-4.5/missing-export/test-missing-export-004.json
#	tests/testthat/_snaps/linux-4.5/missing-export/test-missing-export-005.json
#	tests/testthat/_snaps/linux-4.5/missing-multiple-files/test-missing-multiple-files-001.json
#	tests/testthat/_snaps/linux-4.5/missing-multiple-files/test-missing-multiple-files-002.json
#	tests/testthat/_snaps/linux-4.5/missing-multiple-files/test-missing-multiple-files-003.json
#	tests/testthat/_snaps/linux-4.5/missing-multiple-files/test-missing-multiple-files-004.json
#	tests/testthat/_snaps/linux-4.5/missing-multiple-files/test-missing-multiple-files-005.json
#	tests/testthat/_snaps/linux-4.5/unhashed/test-unhashed-001.json
#	tests/testthat/_snaps/linux-4.5/unhashed/test-unhashed-002.json
#	tests/testthat/_snaps/linux-4.5/unhashed/test-unhashed-003.json
#	tests/testthat/_snaps/linux-4.5/unhashed/test-unhashed-004.json
#	tests/testthat/_snaps/linux-4.5/unhashed/test-unhashed-005.json
#	tests/testthat/_snaps/linux-4.5/wrong-csv-version/test-wrong-csv-version-001.json
#	tests/testthat/_snaps/linux-4.5/wrong-csv-version/test-wrong-csv-version-002.json
#	tests/testthat/_snaps/linux-4.5/wrong-csv-version/test-wrong-csv-version-003.json
#	tests/testthat/_snaps/linux-4.5/wrong-csv-version/test-wrong-csv-version-004.json
#	tests/testthat/_snaps/linux-4.5/wrong-csv-version/test-wrong-csv-version-005.json
#	tests/testthat/_snaps/linux-4.5/wrong-file-type/test-wrong-file-type-001.json
#	tests/testthat/_snaps/linux-4.5/wrong-file-type/test-wrong-file-type-002.json
#	tests/testthat/_snaps/linux-4.5/wrong-file-type/test-wrong-file-type-003.json
#	tests/testthat/_snaps/linux-4.5/wrong-file-type/test-wrong-file-type-004.json
#	tests/testthat/_snaps/linux-4.5/wrong-file-type/test-wrong-file-type-005.json
#	ui.R
#	www/custom.css

mlfurman3 committed:
update system exits logic to use latest period functionality, replaces get_was_lh_info + get_eecr_and_lecr with get_active_info + get_inflows_and_outflows. also kept the Destination column from final data frame

mlfurman3 committed:
remove redundant suppress_values function, since there is a more complete one in system_performance_functions.R

mlfurman3 committed:
create separate tabs for Exits by Year and Exits by Subpopulation in system exits

mlfurman3 committed:
update toggles for system exit download buttons to account for new, separate time chart and subpop chart pages

mlfurman3 committed:
rename "Comparison Group" to "Everyone Else" in exits by subpopulation page

mlfurman3 committed:
update headers above system exit charts: modify race-eth label, add selected subpopulation mini-sub-header

mlfurman3 committed:
convert system exits time and subpop charts to barcharts instead of dotplots

mlfurman3 committed:
split up system exits time subpop downloads into separate pieces for each tab

mlfurman3 committed:
resize system exit time and subpop charts using scale_x_continuous and adjusting spacing and margins in UI

mlfurman3 committed:
reorder detail headers for system exits charts, fix label for exits by subpopulation header

mlfurman3 committed:
rename "System Exit Types" subtab to "Exits by Type" in ui and data downloads

mlfurman3 committed:
update slide names and filenames for system exits by year and by population PPT downloads

mlfurman3 committed:
add black borders to bars in system exit by year, by population barcharts

mlfurman3 committed:
reconfigure system exits by type treemap orientation. now shows Homeless and Permanent on left side, using 2 levels of subgroups

vlopez0603 committed:
System Exits In-App Text

For page instructions, "system exits by type" and "permanent housing demographics" charts. Still tweaking the text for "exits by year" and "exits by subpop" charts.

mlfurman3 committed:
resize System Exits instr header, add line breaks before all chart instructions begin

mlfurman3 committed:
add instructions for System Exits by Year and by Subpopulation tabs, add line breaks throughout all exits instructions

mlfurman3 committed:
add y-axis label to System Exits by Year and by Subpopulation charts, reduce bar widths in PPT export

mlfurman3 committed:
move system exits hex codes into brand.yml file, reference with get_brand_color function

mlfurman3 committed:
Merge branch 'dev' into branch 'feat/system-exits'

Merge branch 'dev' into feat/system-exits
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

mlfurman3 committed:
Merge branch 'dev' into branch 'feat/system-exits'

Merge branch 'dev' into feat/system-exits
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

mlfurman3 committed:
update system exits shinytests

mlfurman3 committed:
additional system exits shinytest updates - dq_main and exportTestValues

mlfurman3 committed:
retain exited_system column from get_inflows_and_outflows for System Exits server, use it to subset filtered data to only rows with exits

mlfurman3 committed:
account for Inactives and other non-permanent exits with NA destination values. uses new fix_missing_destination helper function to recode them to -888 and 99 respectively, and carry this through to data downloads

mlfurman3 committed:
update fix_missing_destination to account for Inactives with permanent destination codes (should count these as Inactive)

mlfurman3 committed:
hide full system overview menu if all enrollments are HP or 07 script fails

mlfurman3 committed:
update logic for computing everyone_else group in system exits subpop data. should now (1) properly account for enrollment filters, (2) filter to correct set of system exits, and (3) apply a special exclusion for children when comparing Veterans to Non-Veterans

mlfurman3 committed:
add args with defaults to date report helper functions, for easier passing of different report periods

mlfurman3 committed:
calculate enrollments for previous year properly, starting with enrollment_categories_prev in 07 script. pass these to new session$userData data frames

mlfurman3 committed:
update system exits by year reactives and data frames using new previous year calcs, which properly increases counts for previous year. these new figures are automatically passed on to the chart, data, and image download

mlfurman3 committed:
update System Overview and System Exits in-app text instructions

mlfurman3 committed:
add demo mode welcome message for System Exits tab, update System Overview message

mlfurman3 committed:
fix everyone else calculation to use correct default for veteran status

mlfurman3 committed:
fix row counts used in System Exit PH Demogrpaphics data download metadata

mlfurman3 committed:
updates to multiple System Exit Chart Overview instructions

mlfurman3 committed:
update chart name in system exit PH demographic instructions

mlfurman3 committed:
update System Performance instructions wrt number of demog filters

mlfurman3 committed:
update time chart previous year calc. add arguments to expand_by_periods, get_active_info to allow for previous year calculations

mlfurman3 committed:
adjust numbered indents for System Overview and System Exit instructions

mlfurman3 committed:
add reportStart and reportEnd args to get_active_info, get_inflows_and_outflows so they are can flexibly work for varying report periods, such as the time chart

mlfurman3 committed:
add separate system exits shinytest function, called within main_test_script. Changes in sys_comp_df appear to all be row reordering

mlfurman3 committed:
add more detailed system exits shinytests - visit each chart tab, apply filters, save outputs

mlfurman3 committed:
add logic for excluding methodology type from System Perf Demographics downloads when it is not directly used in the chart

mlfurman3 committed:
repeat logic change for System Comp chart downloads

mlfurman3 committed:
add dynamic titles to System Exits by Subpopulation and by Year charts

mlfurman3 committed:
For previous year calculation, count as a system exit any client that has an exit on the last day of the previous year report period that does not have a following enrollment within the 14-day period after exit.

mlfurman3 committed:
move system exits shinytests to own testing script, generate initial test JSON files

mlfurman3 committed:
update main-valid shinytests to reflect Sys Overview chart header restoring the Methodology type label

mlfurman3 committed:
additional logging when navigating system exit tabs and downloading reports

mlfurman3 committed:
update system exit data download tab names for consistency

mlfurman3 committed:
add system exits client-level download button and download functionality. uses similar sheets and logic to sys overview one but passes in syse reactives instead of syso ones and only contains first 3 sheets

mlfurman3 committed:
change formulas for % change and % difference in system exits by year and by subpop, integrate counts into tables

mlfurman3 committed:
use map2 for percent change/diff functions in system exit data downloads

vlopez0603 committed:
Update links to HMIS CSV Format Specifications

Updated link on Home and Upload pages

vlopez0603 committed:
Fix typo in Demo System Overview Pop-Up

vlopez0603 committed:
Update Tab and Page Names

- Updated tab names in DQ System and DQ Org pages, removing references to the number 10
- Updated "Glossary" nav option and page title to "System Performance Glossary"

vlopez0603 committed:
Updated DQ Page Instructions

- Made updates to Sys and Org-level page instructions to align with tab name changes

vlopez0603 committed:
Updates to text on DQ pages

- Changed instances of "-wide" to "-level"

vlopez0603 committed:
Updates to demo text

- Ensured that page name in demo pop-ups matched the actual page name

vlopez0603 committed:
Fix typo in Project Descriptor Data Elements text

vlopez0603 committed:
Update to Zero FSA Issues Message

When there are no FSA issues, users should see message encouraging them to explore other pages for DQ and system performance.

mlfurman3 committed:
update time chart validation period to greater than or equal to 729 days

mlfurman3 committed:
add system exits subpop comparison types option - client-level, demographic, or both. update backend data and downloads to reflect currently selected option

mlfurman3 committed:
update shinytest references to top 10 tab names

mlfurman3 committed:
update shinytests to reflect tab name changes and instruction text changes

mlfurman3 committed:
fix typos in System Exits subpopulation demographics option

mlfurman3 committed:
create separate subtab in System Exits page for investigating alternate subpop chart options

mlfurman3 committed:
add logic for hiding bottom row filters, limiting checkboxes to 2 on exits by subpop chart tab

mlfurman3 committed:
add conditional/dynamic inputs that respond to the checkboxes in subpop2 chart tab

mlfurman3 committed:
Merge pull request #980 from abtassociates/link-and-text-updates

Updates to Links, Tab Names, Instructions

mlfurman3 committed:
Update System exits "Permanent Housing Demographics" subtab name to "Exits to PH Demographics" everywhere

mlfurman3 committed:
hide previous subpopulation chart and implement alternative subpopulation chart for system exits: 1-way, 2-way, or 3-way frequency table with selected subpop highlighted in purple, similar style to PH demographics crosstabs

mlfurman3 committed:
reorder categorical count columns so that total is computed first and then used to check for data suppression

mlfurman3 committed:
fix fill scale for PH demographics 1d crosstabs - was improperly replaced

mlfurman3 committed:
fix grouped demog ifelse option for subpopulation chart

mlfurman3 committed:
move subpop conditional dropdowns to ui.R so that resets no longer happen when a variable selection is changed. now need to determine if/when resets should be added back in using updatePickerInput

mlfurman3 committed:
fix orientation of 2x2 subpop chart when demog is on y axis. now uses fct_rev like the other cases

mlfurman3 committed:
separate out subpop checkboxes, always show dropdowns but disable and gray out unless checkbox is checked. there are some changes to reactives as a result, but largely the same logic

mlfurman3 committed:
additional subpop reactive swaps and ensuring that a max of two checkboxes can be checked at once

mlfurman3 committed:
Merge branch 'dev' into branch 'feat/system-exits'

Merge branch 'dev' into feat/system-exits

# Conflicts:
#	tab_instructions.R
#	tests/testthat/_snaps/linux-4.5/fsa/test-fsa-001.json
#	tests/testthat/_snaps/linux-4.5/fsa/test-fsa-002.json
#	tests/testthat/_snaps/linux-4.5/fsa/test-fsa-003.json
#	tests/testthat/_snaps/linux-4.5/fsa/test-fsa-004.json
#	tests/testthat/_snaps/linux-4.5/fsa/test-fsa-005.json
#	tests/testthat/_snaps/linux-4.5/main-valid/test-main-valid-dq-org.json
#	tests/testthat/_snaps/linux-4.5/main-valid/test-main-valid-dq-system.json
#	tests/testthat/_snaps/linux-4.5/main-valid/test-main-valid-exportTestValues.json
#	tests/testthat/_snaps/linux-4.5/main-valid/test-main-valid-just-uploaded.json
#	tests/testthat/_snaps/linux-4.5/missing-export/test-missing-export-001.json
#	tests/testthat/_snaps/linux-4.5/missing-export/test-missing-export-002.json
#	tests/testthat/_snaps/linux-4.5/missing-export/test-missing-export-003.json
#	tests/testthat/_snaps/linux-4.5/missing-export/test-missing-export-004.json
#	tests/testthat/_snaps/linux-4.5/missing-export/test-missing-export-005.json
#	tests/testthat/_snaps/linux-4.5/missing-multiple-files/test-missing-multiple-files-001.json
#	tests/testthat/_snaps/linux-4.5/missing-multiple-files/test-missing-multiple-files-002.json
#	tests/testthat/_snaps/linux-4.5/missing-multiple-files/test-missing-multiple-files-003.json
#	tests/testthat/_snaps/linux-4.5/missing-multiple-files/test-missing-multiple-files-004.json
#	tests/testthat/_snaps/linux-4.5/missing-multiple-files/test-missing-multiple-files-005.json
#	tests/testthat/_snaps/linux-4.5/unhashed/test-unhashed-001.json
#	tests/testthat/_snaps/linux-4.5/unhashed/test-unhashed-002.json
#	tests/testthat/_snaps/linux-4.5/unhashed/test-unhashed-003.json
#	tests/testthat/_snaps/linux-4.5/unhashed/test-unhashed-004.json
#	tests/testthat/_snaps/linux-4.5/unhashed/test-unhashed-005.json
#	tests/testthat/_snaps/linux-4.5/wrong-csv-version/test-wrong-csv-version-001.json
#	tests/testthat/_snaps/linux-4.5/wrong-csv-version/test-wrong-csv-version-002.json
#	tests/testthat/_snaps/linux-4.5/wrong-csv-version/test-wrong-csv-version-003.json
#	tests/testthat/_snaps/linux-4.5/wrong-csv-version/test-wrong-csv-version-004.json
#	tests/testthat/_snaps/linux-4.5/wrong-csv-version/test-wrong-csv-version-005.json
#	tests/testthat/_snaps/linux-4.5/wrong-file-type/test-wrong-file-type-001.json
#	tests/testthat/_snaps/linux-4.5/wrong-file-type/test-wrong-file-type-002.json
#	tests/testthat/_snaps/linux-4.5/wrong-file-type/test-wrong-file-type-003.json
#	tests/testthat/_snaps/linux-4.5/wrong-file-type/test-wrong-file-type-004.json
#	tests/testthat/_snaps/linux-4.5/wrong-file-type/test-wrong-file-type-005.json
#	ui.R
#	upload_server.R

mlfurman3 committed:
Merge branch 'feat/system-exits' into branch 'feat/system-exits-subpop-revamp'

Merge branch 'feat/system-exits' into feat/system-exits-subpop-revamp
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

mlfurman3 committed:
remove no longer used system exits code chunks

mlfurman3 committed:
add to subpop data download and create PPT download, with one slide per destination type

mlfurman3 committed:
re-add group totals to subpop data download, rename percent column in export

mlfurman3 committed:
allow for counts and combinations of 0 exits in subpop chart and data download

mlfurman3 committed:
fix logic for filtering subpop charts and everyone else categories, pass df as arg to chart function

mlfurman3 committed:
fix typo in destination type id

alex-silverman commented on system_exits_server.R:
```diff
@@ -0,0 +1,2153 @@
+    pct_subpop <- count_subpop / .total_s
+    pct_everyone_else <- count_everyone_else / .total_e
+    
+    pct_diff <- purrr::map2_dfr(count_subpop, count_everyone_else, .f = calc_pct_diff, format='num')
```
the documentation says these have been superseded, but I think we can leave as is.

alex-silverman commented on system_exits_server.R:
```diff
@@ -0,0 +1,2153 @@
+    v_total_phd <- plot_df_phd %>%
+      group_by(!!!syms(selections[[1]])) %>%
+      summarise(N = ifelse(all(is.na(n)), NA, sum(n, na.rm = TRUE))) %>%
+      mutate(!!selections[[2]] := 'Total') #%>%
```
`collapse`?

```suggestion
    h_total <- plot_df %>%
        fgroup_by(selections[[2]]) %>%
        fsummarise(N = fsum(n)) %>%
        mutate(!!selections[[1]] := 'Total') %>%
        suppress_values("N") %>%
        suppress_next_val_if_one_suppressed_in_group(selections[1], "N")
      
      v_total <- plot_df %>%
        fgroup_by(selections[[1]]) %>%
        fsummarise(N = fsum(n)) %>%
        mutate(!!selections[[2]] := 'Total') %>%
        suppress_values("N") %>%
        suppress_next_val_if_one_suppressed_in_group(selections[2], "N")
      
      h_total_phd <- plot_df_phd %>%
        fgroup_by(selections[[2]]) %>%
        fsummarise(N = fsum(n)) %>%
        mutate(!!selections[[1]] := 'Total') #%>%
      #suppress_values("N") %>%
      #suppress_next_val_if_one_suppressed_in_group(selections[1], "N")
      
      v_total_phd <- plot_df_phd %>%
        qDT() %>%
        fgroup_by(selections[[1]]) %>%
        fsummarise(N = fsum(n)) %>%
        mutate(!!selections[[2]] := 'Total')
```

Also, this looks really similar to the code in system_composition_server. I'd suggest combining somehow if possible

alex-silverman commented on system_performance_functions.R:
```diff
@@ -0,0 +1,861 @@
+  
+  logMetadata(session, glue("Downloaded System {ttype} Tabular Data: {tabbox}{demotext}", 
+                            ttype=str_to_title(type),
+                            demotext = if_else(isTruthy(T), " - DEMO MODE", "")))
```
I assume this is temporary to add " - DEMO MODE" in all cases?

mlfurman3 replied:
Yep thanks, forgot to switch that back. Just fixed.

alex-silverman commented:
Some architectural suggestions (not that all need to be done):
 - modules for the subtabs and leave core reactives in the system exits server. I know we aren't really using modules in Eva, but this could be a start!
 - split out logic from reactivity and put logic in functions in a separate script
 - the Permanent Housing Demographics logic seems so similar to the system composition logic. If we can combine, we might conserve a whole bunch of space. Maybe like a heatmap-viz function or something.
 - get_syse_compare_time_flextable and get_syse_compare_subpop_flextable look very similar. maybe we can combine?
 - There are 8 total download handlers in this script (4 Excel, 4 PPT) and they all do pretty much the same thing: check metadata, grab the plot/table, format a file name, and execute sys_perf_ppt_export or write_xlsx. Combine?

alex-silverman commented on system_exits_server.R:
```diff
@@ -0,0 +1,2153 @@
+      mutate(!!selections[[2]] := 'Total') #%>%
+    #suppress_values("N") %>%
+    #suppress_next_val_if_one_suppressed_in_group(selections[2], "N")
+  }
```
It's a little unclear everything this whole function is doing. Like we do this `get_sys_plot_df_2vars` call, which I would think would be the main step, followed by the actual plotting. But then we do all this other stuff.

I think we could cut this particular chunk of code in half if not more, as we're doing the same thing to plot_df and plot_df_phd, and within them, using selections[[1]] and selections[[2]].

But we may just want to split things out into functions with names that make it more clear what's happening.

alex-silverman commented on system_exits_server.R:
```diff
@@ -0,0 +1,2153 @@
+    arrange(!!sym(selections[2]))
+  names(export_df)[1:2] <- c(export_label1, export_label2)
+  
+  sys_phd_export(export_df)
```
like this could be inside an `export_2vars_prep` function, or something

alex-silverman commented on system_exits_server.R:
```diff
@@ -0,0 +1,2153 @@
+         how = 'left') %>% 
+    fmutate(frac_export = scales::percent(frac_export, accuracy=0.1)) %>% 
+    fselect(get(selections[2]), get(selections[1]), 'Total Count' = n_orig, 'Permanent Count' = num, 'Percent in Permanent' = frac_export, `Suppression Flag`) %>% 
+    arrange(!!sym(selections[2]))
```
does `collapse::roworder` not work here?

mlfurman3 replied:
updated to `roworderv` which can handle the named vector. apparently that's the difference between `roworder` and `roworderv`

alex-silverman commented on system_exits_server.R:
```diff
@@ -0,0 +1,2153 @@
+  sys_phd_export(export_df)
+  
+  if(methodology_type == 1){
+    h_total_joined <- join(h_total,h_total_phd %>% frename(num = N), how = 'left', on=c(selections[1],selections[2])) %>% 
```
```suggestion
    h_total_joined <- join(h_total,h_total_phd %>% frename(num = N), how = 'left', on=selections) %>% 
```

mlfurman3 committed:
show unsuppressed values in subpop data download, show 0% for 0/0 cases

mlfurman3 committed:
change 1d subpop chart to columns instead of rows

mlfurman3 committed:
improve alignment of subpop checkboxes and dropdowns to match their similar filters from other tabs

alex-silverman commented on system_composition_server.R:
```suggestion
h_total2 <- plot_df %>%
  fgroup_by(selections[2]) %>%
  fsummarise(N = fsum(n))
h_total2[[selections[1]]] <- "Total"
```

mlfurman3 committed:
extend widths of subpop dropdowns for larger screen sizes to max of their containers

mlfurman3 committed:
update glossary title

mlfurman3 committed:
escape HTML characters in in-app glossary table

mlfurman3 committed:
glossary download: remove line wrapping, apply font, bold headers, and autofit to width

mlfurman3 committed:
apply css width to method 2 demographic filter in subpop tab

mlfurman3 committed:
specify levels in 3-way subpop chart to ensure HH type is outer horizontal var

mlfurman3 committed:
correct input references used in subpop data download metadata sheet

mlfurman3 committed:
update exits by subpopulation tab name to remove the "2"

mlfurman3 committed:
update System Exits grouped demog label when methodology type switches

mlfurman3 committed:
remove unused server chunks from previous subpop chart configuration

mlfurman3 committed:
fix metadata logging logic check

mlfurman3 committed:
Update system_exits_server.R

Co-authored-by: alex-silverman <alexsilverman96@gmail.com>

mlfurman3 committed:
change 2D system comp heatmap logic to collapse

Co-authored-by: alex-silverman <alexsilverman96@gmail.com>

mlfurman3 committed:
correction to prior commit - convert system comp 2D code to collapse

mlfurman3 committed:
update two arranges to collapse roworderv

mlfurman3 committed:
additional unused code cleanup

mlfurman3 committed:
replace all instances of 'subpop2' with 'subpop' for system exits

mlfurman3 committed:
remove unused UI code for old subpop chart

mlfurman3 committed:
Merge pull request #988 from abtassociates/feat/system-exits-subpop-revamp

Remake of System Exits Subpop Chart

mlfurman3 committed:
modify glossary special characters in app and download, update PDF

mlfurman3 committed:
update glossary PDF in www folder

mlfurman3 committed:
update glossary instructions

mlfurman3 committed:
Merge branch 'glossary-table-updates' into 'feat/system-exits'

Merge branch 'glossary-table-updates' into feat/system-exits
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

mlfurman3 committed:
rename glossary pdf download

mlfurman3 committed:
update text instructions for subpop chart checkboxes

mlfurman3 committed:
correct typo in subtab class to allow right placement of glossary download btn

mlfurman3 committed:
use bslib card for glossary container instead of subtab, ensure download button is right aligned

mlfurman3 committed:
move system exits ui to its own separate script

mlfurman3 committed:
remove outdated system exits server reactives for subpop tab

mlfurman3 committed:
update everyone_else variable name

mlfurman3 committed:
refactor system_exits_server.R into separate scripts for each subtab

mlfurman3 committed:
amend and display title for system exits subpop chart. logic was already in place to pull counts for each group

mlfurman3 committed:
amend and display title for subpop chart. logic was already in place to pull counts

mlfurman3 committed:
add subheader with selection and date info to subpopulation subtab. works somewhat differently from other tabs due to independent checkboxes

mlfurman3 committed:
update system exit subtab instructions for types, time, and subpop charts

mlfurman3 committed:
move some generic useful functions from system exits to helper_functions.R

mlfurman3 committed:
cleanup and standardize some system exits function names for data, charts, and tables

mlfurman3 committed:
implement new add_destination_type helper function throughout system exit scripts

mlfurman3 committed:
Merge dev into 'feat/system-exits'

Merge branch 'dev' into feat/system-exits

# Conflicts:
#	glossary.R
#	server.R
#	tests/testthat/_snaps/linux-4.5/fsa/test-fsa-001.json
#	tests/testthat/_snaps/linux-4.5/fsa/test-fsa-002.json
#	tests/testthat/_snaps/linux-4.5/fsa/test-fsa-003.json
#	tests/testthat/_snaps/linux-4.5/fsa/test-fsa-004.json
#	tests/testthat/_snaps/linux-4.5/fsa/test-fsa-005.json
#	tests/testthat/_snaps/linux-4.5/main-valid/test-main-valid-just-uploaded.json
#	tests/testthat/_snaps/linux-4.5/missing-export/test-missing-export-001.json
#	tests/testthat/_snaps/linux-4.5/missing-export/test-missing-export-002.json
#	tests/testthat/_snaps/linux-4.5/missing-export/test-missing-export-003.json
#	tests/testthat/_snaps/linux-4.5/missing-export/test-missing-export-004.json
#	tests/testthat/_snaps/linux-4.5/missing-export/test-missing-export-005.json
#	tests/testthat/_snaps/linux-4.5/missing-multiple-files/test-missing-multiple-files-001.json
#	tests/testthat/_snaps/linux-4.5/missing-multiple-files/test-missing-multiple-files-002.json
#	tests/testthat/_snaps/linux-4.5/missing-multiple-files/test-missing-multiple-files-003.json
#	tests/testthat/_snaps/linux-4.5/missing-multiple-files/test-missing-multiple-files-004.json
#	tests/testthat/_snaps/linux-4.5/missing-multiple-files/test-missing-multiple-files-005.json
#	tests/testthat/_snaps/linux-4.5/unhashed/test-unhashed-001.json
#	tests/testthat/_snaps/linux-4.5/unhashed/test-unhashed-002.json
#	tests/testthat/_snaps/linux-4.5/unhashed/test-unhashed-003.json
#	tests/testthat/_snaps/linux-4.5/unhashed/test-unhashed-004.json
#	tests/testthat/_snaps/linux-4.5/unhashed/test-unhashed-005.json
#	tests/testthat/_snaps/linux-4.5/wrong-csv-version/test-wrong-csv-version-001.json
#	tests/testthat/_snaps/linux-4.5/wrong-csv-version/test-wrong-csv-version-002.json
#	tests/testthat/_snaps/linux-4.5/wrong-csv-version/test-wrong-csv-version-003.json
#	tests/testthat/_snaps/linux-4.5/wrong-csv-version/test-wrong-csv-version-004.json
#	tests/testthat/_snaps/linux-4.5/wrong-csv-version/test-wrong-csv-version-005.json
#	tests/testthat/_snaps/linux-4.5/wrong-file-type/test-wrong-file-type-001.json
#	tests/testthat/_snaps/linux-4.5/wrong-file-type/test-wrong-file-type-002.json
#	tests/testthat/_snaps/linux-4.5/wrong-file-type/test-wrong-file-type-003.json
#	tests/testthat/_snaps/linux-4.5/wrong-file-type/test-wrong-file-type-004.json
#	tests/testthat/_snaps/linux-4.5/wrong-file-type/test-wrong-file-type-005.json
#	ui.R
#	www/custom.css

mlfurman3 committed:
Merge branch 'feat/system-exits' into 'feat/system-exits-module'

Merge branch 'feat/system-exits' into feat/system-exits-module

# Conflicts:
#	system_exits_server.R

mlfurman3 committed:
fix typo from merge conflict

mlfurman3 committed:
minor instructions HTML formatting

mlfurman3 committed:
remove extra browser statements

mlfurman3 committed:
Merge pull request #995 from abtassociates/feat/system-exits-module

System Exits script refactoring

mlfurman3 committed:
update percent change formula in system exit instructions

mlfurman3 committed:
code reorg within system_exits_server script

mlfurman3 committed:
correct subpop HH type discrepancies - logic was adjusted for complements of subpop Household type to correctly compute the everyone_else group in each case

mlfurman3 committed:
remove dash from HH type selection in subpop chart headers

mlfurman3 committed:
fix returning of subheaders for all system performance charts

mlfurman3 committed:
adjust subpop logic to avoid double counting for non-HH-type filter scenarios

mlfurman3 committed:
reorder and indent subpop chart subheaders for demographics selections

mlfurman3 committed:
correct system performance subheader logic for updating based on methodology type. previously was not switching between method 1 and method 2 correctly

mlfurman3 committed:
only indent subheaders for system exits subpop page, exclude others like sys overview

mlfurman3 committed:
update system exits shinytest script to reflect subpop filter and chart changes

mlfurman3 committed:
regenerate shinytests with latest system exits edits and test script

mlfurman3 committed:
additional main-valid shinytest updates

mlfurman3 committed:
move system exit shinytest functions to testing_functions.R to allow GHA to run properly

mlfurman3 committed:
correct bad input id reference in system exits shinytests

mlfurman3 committed:
add idle time to system exits by type shinytest

mlfurman3 committed:
hide glossary download button

mlfurman3 committed:
add exportTestValues call for system exits by type

mlfurman3 committed:
push minor shinytest edit

mlfurman3 committed:
update exportTestValues shinytest file

Commit:
'CLE Instructions and Data Dictionary.xlsx'

# Upload as new file in '/www'
- contains revised Data Dictionary along with formatted Instructions sheet

client-level-export-data-dictionary.csv

# Save sheet from excel file as a csv
- upload differences to csv file in '/www'
- clean up strings that got converted to '?'

Commit:
client_level_export_server.R

# update filter_selections (Metadata sheet)
- to set the column names according to the first row of data
- subset the first row out of the dataset, so it is not duplicated

# update building client_level_export_list
- to load new item 'instructions' as the "Instructions" sheet of the 'CLE Instructions and Data Dictionary.xlsx'
- update data_dictionary item to load from the same workbook but sheet "Data Dictionary
- update names to label the new item "Instructions" and rename "Adjusted Non-Res" to "Adjusted Enrollments"

# Add commented out re-write attempt that uses Officer Package
- this would preserve the formatting of the original excel file (Instructions sheet) but has some issues explained by code comments

sarasoko committed:
Merge branch '1009-update-download-tabs' into feat/system-exits

mlfurman3 committed:
Update changelog with system exits and zero util changes

mlfurman3 committed:
Remove tabSystemExits duplicate ui script, adjust logging statements. Will handle separating ui scripts in future branch
